tests/rptest/tests/full_disk_test.py

```diff
@@ -642,5 +650,10 @@ def target_size_reached():
                 )
             return below
 
-        # give it plenty of time. on debug it is hella slow
-        wait_until(target_size_reached, timeout_sec=120, backoff_sec=5)
+        if cleanup_policy == TopicSpec.CLEANUP_COMPACT:
+            # For `cleanup.policy=compat` we don't expect any removal. Wait for timeout.
+            with expect_exception(TimeoutError, bool):
+                wait_until(target_size_reached, timeout_sec=120, backoff_sec=5)
```

Comment:
I think it'd be a stronger check to have this enforce that the local start offset is always 0.

Reply:
nice idea